### DISCUSSION TITLE
this script needs internationalization for easier access

### DISCUSSION THREAD

**Discussion by surepy** [ThioJoe/YT-Spammer-Purge#617] [Ideas]
Not everyone knows how to speak English, and that goes for all the content creators in youtube too.
I propose that by using a library like [python-18n](https://pypi.org/project/python-i18n/) we make it friendlier for everyone around the world.

this script works *shockingly* well, like the differences between channels that use it and don't use it are pretty significant in my personal experience; and I just want more people to use it and make youtube a safer place.

**Comment by ThioJoe** [comment]
Hm I'd like to see some examples of other scripts using this. I wouldn't mind implementing a system that would make it easy for people to make pull requests to add their own language and stuff. The biggest concern is considering all the displayed text is currently hard coded, I assume I'd have to replace it all with references to language files, and put all the writing into those? That would mean going into all the menus and changing them up

**Reply by surepy** [reply]
I haven't found other scripts using that exact library- it was just the first googled result and I felt like I needed an example.
it really shouldn't matter whichever library is used, as the process should be pretty similar across the board.

> I assume I'd have to replace it all with references to language files, and put all the writing into those? That would mean going into all the menus and changing them up

yes. we will have to replace all the strings to something like ``print("message {variable}")`` to ``print(_("translation_key", argument_1=variable))`` for example.

**Reply by Kurdakov** [reply]
usually it's gettext so it looks like
import gettext
_ = gettext.gettext
then
print(_("This is a translatable string"))
or
output = _('Today is %(month)s %(day)s.') % {'month': m, 'day': d}

so one possible link https://phrase.com/blog/posts/translate-python-gnu-gettext/

**Comment by TechStudent10** [comment]
You might want to look at [martinet101/ElevenClock](https://github.com/martinet101/ElevenClock). It allows translations.

This is just an example of a project that uses translations.
